### ISSUE TITLE
Flexible Field Handling

### ISSUE THREAD

**Issue by StalkerRaftik** [nimaxin/init-data-py#11]
This library is not suitable for use in a production-ready environment. Whenever a new field appears in init_data, both validation and data parsing immediately break. I believe the library should be redesigned to handle new fields introduced by Telegram gracefully. Instead of returning objects, it should return a parsed dictionary with fields, leaving out validation (which it generally shouldn’t be responsible for).

**Comment by nimaxin** [comment]
You’re right; this library isn’t production-ready yet. I’m currently working on version 1, which brings many changes to address these issues and fix the design flaws. In the new version, the validation logic for init_data is completely separated from parsing, and validation is implemented based on Telegram’s official guidelines to ensure it always works as expected.

Regarding the poor design of the current version, I agree with you; this has been a key focus for improvement in v1. I’m planning to release it either tonight or tomorrow night.

**Comment by nimaxin** [comment]
Fixed in 1.0.0. Unknown fields are kept and readable through `init_data.extra`
instead of breaking parsing, and parsing is now separate from validation:
`parse()` reads, `validate_by_hash()` checks.
